### PR TITLE
Testing cosmos

### PR DESCRIPTION
**Pull request by kuehma02**
I followed the videos from this Azure Tutorial (https://docs.microsoft.com/en-us/azure/cosmos-db/tutorial-develop-mongodb-react) to connect to the Cosmos DB

I also did some restructuring and renaming of some of the files. 

FINALLY HOSTED!!! 😄

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
+    // Use IntelliSense to learn about possible Node.js debug attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+    "version": "0.2.0",
+    "compounds": [
```
Comment (rublan01) [inline .vscode/launch.json]: Could we add the .vscode/ folder to our .gitignore? :)

File: server/server.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+  if (isNaN(port)) {
+    // named pipe
+    return val;
+  }
+
+  if (port >= 0) {
+    // port number
+    return port;
+  }
```
Comment (rublan01) [inline server/server.js]: This looks good - just curious: is this how you bypassed the host header/port errors? Like is this just a way for us to manage and add checks around the ports that we're using ?

File: src/components/EditUser.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    return (
+      <div>
+        <div className="editfields">
+          <div>
+            <label>id: </label>
+            {props.addingUser
+              ? <input
+                  type="number"
+                  name="id"
+                  placeholder="id"
```
Comment (rublan01) [inline src/components/EditUser.js]: Eventually we probably don't want our users to add their own 'id' :) But this is fine for now!

File: src/components/EditUser.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+              onChange={props.onChange}
+            />
+          </div>
+          <div>
+            <label>password: </label>
+            <input
+              name="password"
+              value={props.selectedUser.password}
+              placeholder="password"
+              onChange={props.onChange}
```
Comment (rublan01) [inline src/components/EditUser.js]: Along that same vein, we won't want the password to be publicly viewable :) But that can be updated later! :)

File: src/components/User.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+            <button
+                className="delete-button"
+                onClick={e => props.onDelete(e, props.user)}
+            >
+                Delete
+            </button>
+            <div className="user-element">
+                <div className="badge">{props.user.id}</div>
+                <div className="name">{props.user.name}</div>
+                <div className="password">{props.user.password}</div>
```
Comment (rublan01) [inline src/components/User.js]: This is where we could take out showing the password if we really wanted to haha

File: src/components/Users.js [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+                        return <User 
+                            user={user} 
+                            onSelect={this.handleSelect} 
+                            selectedUser = {this.state.selectedUser}
+                            onDelete={this.handleDelete} 
+                        />
+                    })}
+                </ul>
+                <div className="editarea">
+                    <button onClick={this.handleEnableAddMode}>Add New Hero</button>
```
Comment (rublan01) [inline src/components/Users.js]: Change to User?

File: server/server.js (hunk 2 above)
Comment (kuehma02) [reply server/server.js]: Honestly, I got this from the the tutorial code! I think it might be though.

**Pull request merged**